source/opt/loop_unroller.cpp

```diff
@@ -286,6 +286,10 @@ class LoopUnrollerUtilsImpl {
   // to be the actual value of the phi at that point.
   void LinkLastPhisToStart(Loop* loop) const;
 
+  // Kill DebugDeclare or DebugValue that is used for the declaration with
+  // Deref from |bb|.
```

Comment:
What does 'DebugValue that is used for the declaration' mean?  May be: "Kill all debug declaration instructions from |bb|".  It seems that there are two ways to tie debug info to a variable:

* `DebugDeclare`
* `DebugValue` with a dereference expression.  Used when a compound type is scalarized (e.g. by SROA).

We seem to be overloading terminology.  Perhaps call these something like "debug info declaration" instructions?  It uses the word "declare" again, but I can't think of anything better atm.  It should convey that there are more than one way to associate debug info with a memory location or part of a memory location.  Names are hard ;)

Reply:
"Kill all debug declaration instructions from |bb|" sounds good. Thanks!